datacapture/src/main/java/com/google/android/fhir/datacapture/validation/ValueConstraintValidator.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.datacapture.validation
+
+import org.hl7.fhir.r4.model.Questionnaire
+import org.hl7.fhir.r4.model.QuestionnaireResponse
+
+internal open class ValueConstraintValidator(
+  val url: String,
+  val predicate: (Int, Int) -> Boolean,
```

Comment:
hm sorry i missed this in the last review - I think the predicate should be type agnostic. If you keep it this way, in the next PR you'll need to write a separate validator for each data type for max and min. that's too much.

so you should pass in the Type (which is the value in the extension) and the QuestionnaireResponseItemAnswerComponent which is the value in the answer. and in the predicate in the max and min validators you should have a when statement that compares each type.

```suggestion
  val predicate: (Type, QuestionnaireResponseItemAnswerComponent) -> Boolean,
```

Reply:
@jingtang10 Using "Extension" instead of "Type"